AppController/djinn.rb

```diff
@@ -1289,10 +1289,15 @@ def update_node_info_cache()
             next
           end
         end
-        new_stats << node_stats
+        Thread.current[:node_stats] = node_stats
       }
-      @cluster_stats = new_stats
     }
+    new_stats = []
+    threads.each do |t|
+      t.join
```

Comment:
The join here will make the AC on the shadow to wait for the response, and if one of the node is down, it could also incur in some timeouts. If you do so, I would suggest to add a lock at the beginning, and skip the update if a previous update is running.